virtual_undercloud.yml

```diff
@@ -0,0 +1,104 @@
+---
+- hosts: localhost
+  tasks:
+    - include_tasks: tasks/copykeys.yml
+      vars:
+        hostname: "localhost"
+        ssh_user: "root"
+
+    - name: clean previous vm
+      shell: |
+         source {{ infrared_dir }}/.venv/bin/activate
+         infrared virsh --host-address 127.0.0.1 --host-key /root/.ssh/id_rsa --topology-network virtual_uc_network --topology-nodes virtual_uc_topology:1 --host-memory-overcommit True --cleanup True
+      retries: 3
+      ignore_errors: yes
+
+    - name: create undercloud vm
+      shell: |
+         source {{ infrared_dir }}/.venv/bin/activate
+         infrared virsh --host-address 127.0.0.1 --host-key /root/.ssh/id_rsa --topology-network virtual_uc_network --topology-nodes virtual_uc_topology:1 --host-memory-overcommit True -v 2>&1 | tee log_virsh1
+
+    - include_tasks: tasks/get_interpreter.yml
+      vars:
+        hostname: "{{ undercloud_hostname }}"
+        user: "cloud-user"
+
+    - name: python interpreter
+      set_fact:
+        python_interpreter: "{{ (python_version.stderr_lines|length > 0) | ternary('/usr/libexec/platform-python', '/usr/bin/python') }}"
+
+    - name: add undercloud_host to inventory file
+      add_host:
+        name: "undercloud"
+        ansible_host: "{{ undercloud_hostname }}"
+        ansible_ssh_private_key_file: "{{ ansible_ssh_key }}"
+        ansible_user: "cloud-user"
+        ansible_python_interpreter: "{{ python_interpreter }}"
+
+- hosts: undercloud
+  vars:
+    user: "stack"
+  gather_facts: yes
+  tasks:
+    - name: make sure line 'dns=none' is set in /etc/NetworkManager/NetworkManager.conf
```

Comment:
This is nice work masco (as I know the context why you are playing with network manager).
Can you please add comment why you are doing this.

Reply:
done